### ISSUE TITLE
Issue in using tfio.experimental.filter.sobel in Keras model.fit with Colab TPU

### ISSUE THREAD

**Issue by vinits5** [tensorflow/tensorflow#50403]
Hello,

I am training a GAN for image inpainting. I am computing the Sobel edges of my generator's output and giving them as an input to the discriminator. 
1. I am using tfio.experimental.filter.sobel function.
2. I have also tried tf.image.sobel_edges but it is not compatible with TPU.
2. I have defined this as a tf.keras.Model class and then doing the training using model.fit function.
3. I am training this on Google Colab's TPU.

Whenever, I give the sobel edges of the generator's output to the discriminator, the training loss goes to "NaN". I have checked that the computed sobel edges don't contain any NaN and my dataset also doesn't contain any NaN. If I don't give this sobel edges as an input to discriminator then the code works fine without leading to NaN.

Has anyone faced any such similar situation before? Please let me know how can I debug it. If you have faced something similar then, please let me know the solution. Any help is appreciated. Thank you.

**Comment by tilakrayal** [comment]
@vinits5 ,

Please refer this link for information on sobel images.[link1](https://www.tensorflow.org/api_docs/python/tf/image/sobel_edges),[link2](https://www.tensorflow.org/io/api_docs/python/tfio/experimental/filter/sobel).
Also,In order to expedite the trouble-shooting process, could you please provide the complete code and dataset to reproduce the issue reported here.
